pkg/cache/cache.go

```diff
@@ -170,6 +170,12 @@ type Options struct {
 	// instead of `reconcile.Result{}`.
 	SyncPeriod *time.Duration
 
+	// WatchTimeoutPeriod is the timeout for the watch request. If the watch request
+	// times out, the cache will close the watch and reconnect.
+	//
+	// Defaults to a random duration between 5 and 10 minutes if unset.
+	WatchTimeoutPeriod *time.Duration
```

Comment:
LG. Will it be easier to understand if name it like `RewatchPeriod`?

A `WatchTimeoutPeriod` field in `cache.Options` seems to be the timeout that waits for the cache starting to watch?

WDYT @xiang90 @sbueringer @vincepri